File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/file/fakes/FakePreparedPlayableFile.java
package com.lasthopesoftware.bluewater.client.playback.file.fakes;

import com.lasthopesoftware.bluewater.client.playback.file.NoTransformVolumeManager;
import com.lasthopesoftware.bluewater.client.playback.file.PlayableFile;
import com.lasthopesoftware.bluewater.client.playback.file.buffering.IBufferingPlaybackFile;
import com.lasthopesoftware.bluewater.client.playback.file.preparation.PreparedPlayableFile;


public class FakePreparedPlayableFile<PlaybackHandler extends PlayableFile & IBufferingPlaybackFile> extends PreparedPlayableFile {

	public FakePreparedPlayableFile(PlaybackHandler playbackHandler) {
		super(playbackHandler, new NoTransformVolumeManager(), playbackHandler);
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/file/fakes/FakePreparedPlayableFile.kt
package com.lasthopesoftware.bluewater.client.playback.file.fakes

import com.lasthopesoftware.bluewater.client.playback.file.NoTransformVolumeManager
import com.lasthopesoftware.bluewater.client.playback.file.PlayableFile
import com.lasthopesoftware.bluewater.client.playback.file.buffering.BufferingPlaybackFile
import com.lasthopesoftware.bluewater.client.playback.file.preparation.PreparedPlayableFile

class FakePreparedPlayableFile<PlaybackHandler>(playbackHandler: PlaybackHandler) :
    PreparedPlayableFile(
        playbackHandler,
        NoTransformVolumeManager(),
        playbackHandler
    ) where PlaybackHandler : PlayableFile, PlaybackHandler : BufferingPlaybackFile
